scripts/template_upload_all.py

```diff
@@ -62,102 +62,130 @@ def parse_cmd_line():
     return args
 
 
-def template_name(image_link, image_ts, checksum_link, version=None):
+def template_name(image_link, timestamp, version=None):
+    """Generate a template name from given link, a timestamp, and optional version string
+    This method should handle naming templates from the following URL types:
+        - http://file.cloudforms.lab.eng.rdu2.redhat.com/builds/manageiq/master/latest/
+        - http://file.cloudforms.lab.eng.rdu2.redhat.com/builds/manageiq/gaprindashvili/stable/
+        - http://file.cloudforms.lab.eng.rdu2.redhat.com/builds/manageiq/fine/stable/
+        - http://file.cloudforms.lab.eng.rdu2.redhat.com/builds/cfme/5.7/stable/
+        - http://file.cloudforms.lab.eng.rdu2.redhat.com/builds/cfme/5.9/latest/
+
+    These builds fall into a few categories:
+        - MIQ nightly (master/latest)  (upstream)
+        - MIQ stable (<name>/stable)  (upstream_stable, upstream_fine, etc)
+        - CFME nightly (<stream>/latest)  (downstream-nightly)
+        - CFME stream (<stream>/stable)  (downstream-<stream>)
+
+    The generated template names should follow the syntax with 5 digit version numbers:
+        - MIQ nightly: miq-nightly-<yyyymmdd>  (miq-nightly-201711212330)
+        - MIQ stable: miq-stable-<name>-<number>-yyyymmdd  (miq-stable-fine-4-20171024)
+        - CFME nightly: cfme-nightly-<version>-<yyyymmdd>  (cfme-nightly-59000-20170901)
+        - CFME stream: cfme-<version>-<yyyymmdd>  (cfme-57402-20171202)
+
+    Release names for upstream will be truncated to 5 letters (thanks gaprindashvili...)
+    """
+
     pattern = re.compile(r'.*/(.*)')
-    image_name = pattern.findall(image_link)[0]
-    image_name = image_name.lower()
-    image_dt = get_last_modified(checksum_link)
-    # CFME brew builds
-    if CFME_BREW_ID in image_name:
+    image_name = pattern.findall(image_link)[0].lower()
+    formatted_timestamp = '{year}{month}{day}'.format(year=str(timestamp.year),
+                                                      month=str(timestamp.month).zfill(2),
+                                                      day=str(timestamp.day).zfill(2))
+    # CFME
+    if CFME_ID in image_name:
         if 'nightly' in image_name:
-            # 5.6+ nightly
```

Comment:
As best as I can tell, this code was built to handle an older directory/build structure/naming convention, and updated bit-by-bit to handle newer structure/naming and still supporting the older.

This has been tested against all of the trigger URLs used by CI that call this script.